src/libs/actions/Task.js

```diff
@@ -0,0 +1,139 @@
+
+    // Optimistically created Action on the parent
+    const optimisticCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(parentReportID);
+
+    // Create the task report
+    const optimisticTaskReport = ReportUtils.buildOptimisticTaskReport(currentUserEmail, assignee, parentReportID, optimisticCreatedAction.reportActionID, title, description);
+
+    // Create the CreatedReportAction on the task
+    const optimisticTaskCreatedAction = ReportUtils.buildOptimisticCreatedReportAction(optimisticTaskReport.reportID);
+
+    // AddCommentReportAction on the parent chat report
+    const AddCommentText = ` created a task: ${title}`;
+    Report.addComment(parentReportID, AddCommentText);
+
+    const optimisticData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.SET,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: optimisticTaskReport,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: optimisticTaskCreatedAction},
+        },
+    ];
+
+    const successData = [];
+
+    const failureData = [
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT}${optimisticTaskReport.reportID}`,
+            value: null,
+        },
+        {
+            onyxMethod: CONST.ONYX.METHOD.MERGE,
+            key: `${ONYXKEYS.COLLECTION.REPORT_ACTIONS}${optimisticTaskReport.reportID}`,
+            value: {[optimisticTaskCreatedAction.reportActionID]: {pendingAction: null}},
+        },
+    ];
+
+    API.write(
+        'CreateTask',
+        {
+            // parentReportActionID: optimisticCreatedAction.reportActionID,
+            parentReportActionID: '',
+            parentReportID,
+            taskReportID: optimisticTaskReport.reportID,
+            title,
+            description,
+            assignee,
+            assigneeChatReportID,
+        },
+        {optimisticData, successData, failureData},
+    );
+
+    clearOutTaskInfo();
+
+    Navigation.navigate(ROUTES.getReportRoute(optimisticTaskReport.reportID));
+}
+
+function setDetailsValue(title, description) {
+    // This is only needed for creation of a new task and so it should only be stored locally
+    Onyx.merge(ONYXKEYS.TASK, {title, description});
+}
+
+function setTitleValue(title) {
+    Onyx.merge(ONYXKEYS.TASK, {title});
+}
+
+function setDescriptionValue(description) {
+    Onyx.merge(ONYXKEYS.TASK, {description});
+}
+
+function setAssigneeValue(assignee) {
+    let newChat = {};
+    const chat = ReportUtils.getChatByParticipants([assignee]);
+    if (!chat) {
+        newChat = ReportUtils.buildOptimisticChatReport([assignee]);
+    }
+    const reportID = chat ? chat.reportID : newChat.reportID;
+
+    Report.openReport(reportID, [assignee], newChat);
+
```

Comment:
Here is where you'd check if we already have a share destination.  If not, set it to the report between the user and assignee